app/src/recipes/recipes.controller.js

```diff
@@ -14,23 +14,39 @@ class RecipesCtrl {
 
 	//TODO: persist state of checkboxes in service and re-initialize when setting view
 
+	//TODO: if any of the filters get applied, need to rerun the others...
 	filterByCategory(category) {
-		console.log(this.selectedCategories);
+		let selectedCategories = _.chain(this.selectedCategories)
```

Comment:
write one filter function, that takes everything into account -- search, checkboxes, veg flag

bail out early if certain conditions are met to keep it fast